lean4/src/putnam_1963_a3.lean

```diff
@@ -1,21 +1,17 @@
 import Mathlib
 open BigOperators
 
-open Topology Filter
+open Nat Set Topology Filter
 
 noncomputable abbrev putnam_1963_a3_solution : (ℝ → ℝ) → ℕ → ℝ → ℝ → ℝ := sorry
--- fun (f : ℝ → ℝ) (n : ℕ) (x : ℝ) (t : ℝ) ↦ (x - t)^(n - 1) * (f t) / (Nat.factorial (n - 1) * t^n)
+-- fun (f : ℝ → ℝ) (n : ℕ) (x : ℝ) (t : ℝ) ↦ (x - t) ^ (n - 1) * (f t) / ((n - 1)! * t ^ n)
 theorem putnam_1963_a3
-(n : ℕ)
-(f : ℝ → ℝ)
-(P D : ℕ → (ℝ → ℝ) → (ℝ → ℝ))
-(δ : (ℝ → ℝ) → (ℝ → ℝ))
-(hδ : δ = fun g : ℝ → ℝ ↦ (fun x : ℝ ↦ x) * deriv g)
-(hD : D = fun (m : ℕ) (g : ℝ → ℝ) ↦ δ g - (fun x : ℝ ↦ (m : ℝ)) * g)
-(y : ℝ → ℝ)
-(hy : y = fun x : ℝ ↦ ∫ t in Set.Ioo 1 x, putnam_1963_a3_solution f n x t)
-(hn : n ≥ 1)
-(hf : Continuous f)
-(hP : P 0 y = y ∧ ∀ m ∈ Finset.range n, P (m + 1) y = D (n - 1 - m) (P m y))
-: (ContDiff ℝ n y) ∧ (∀ x ≥ 1, P n y x = f x) ∧ (∀ i ∈ Finset.range n, iteratedDeriv i y 1 = 0) :=
-sorry
+    (P : ℕ → (ℝ → ℝ) → (ℝ → ℝ))
+    (hP : P 0 = id ∧ ∀ i y, P (i + 1) y = P i (fun x ↦ x * deriv y x - i * y x))
+    (n : ℕ)
+    (hn : 0 < n)
+    (f y : ℝ → ℝ)
+    (hf : ContinuousOn f (Ici 1)) :
+    ContDiffOn ℝ n y (Ici 1) ∧ (∀ i < n, deriv^[i] y 1 = 0) ∧ P n y = f ↔
```

Comment:
Should it not be that P n y and f are equal on [1, \infty)? I am not sure it matters either way but may be closer to the informal statement.

Reply:
As you rightly point out, technically it doesn't matter. However I tend to agree so I have made this change.

Reply:
Actually I now realise your suggestion is necessary for the converse direction to be true: good catch!